pkg/apply/merge.go

```diff
@@ -76,6 +80,169 @@ func MergeDeploymentForUpdate(current, updated *uns.Unstructured) error {
 	return nil
 }
 
+// MergeDaemonSetForUpdate preserves ptp-security-volume added by ptpconfig_controller.
+// This volume is dynamically managed based on PtpConfig CRs and should not be
+// overwritten when PtpOperatorConfig reconciles the DaemonSet template.
+func MergeDaemonSetForUpdate(current, updated *uns.Unstructured) error {
```

Comment:
This function should simply merge the 2 daemonset templates, specifically the volumes.